wasmapi/rust/src/rust_bindings/syscall.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::mem;
+
+use crate::rust_bindings::helpers::{bytes_to_buf_ptr, from_c_string, string_to_buf_ptr};
+
+extern "C" {
+    #[link_name = "getSyscallName"]
+    fn get_Syscall_name(id: u32, dst: u64) -> u32;
+    #[link_name = "getSyscallID"]
+    fn get_Syscall_id(name: u64) -> i32;
+    #[link_name = "getSyscallDeclaration"]
+    fn get_Syscall_declaration(name: u64, pointer: u64) -> u32;
+}
+
+pub const IS_POINTER_FLAG: u32 = 1;
+pub const MAX_SYSCALL_LENGTH: usize = 64;
+
+pub type Result<T> = std::result::Result<T, ErrSyscall>;
+
+pub enum ErrSyscall {
+    ErrSysName(String),
+    ErrSysID(String),
+    ErrDecSys(String),
+}
+
```

Comment:
Do we really need three diffent error kinds? Can't we use `String` for all of them?